pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -107,6 +124,56 @@ func (s secretResource) computeSecret(identities []byte) *corev1.Secret {
 		Type:       corev1.SecretTypeOpaque,
 		StringData: map[string]string{consts.ServerIdentitiesFilename: string(identities)},
 	}
-	controllerutil.SetControllerReference(s.infinispan, secret, s.scheme)
-	return secret
+
+	s.log.Info(fmt.Sprintf("Creating Identities Secret %s", s.infinispan.GetSecretName()))
+	_, err = controllerutil.CreateOrUpdate(ctx, s.client, secret, func() error {
+		return controllerutil.SetControllerReference(s.infinispan, secret, s.scheme)
+	})
+
+	if err != nil {
+		return fmt.Errorf("Unable to create user identities secret: %w", err)
+	}
+	return nil
+}
+
+func (s secretResource) createAdminIdentitiesSecret(namespace string) error {
+	identities, err := users.GetAdminCredentials()
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.AdminSecretName,
+			Namespace: namespace,
+			Labels:    infinispan.LabelsResource(s.infinispan.Name, "infinispan-secret-admin-identities"),
+		},
+		Type:       corev1.SecretTypeOpaque,
+		StringData: map[string]string{consts.ServerIdentitiesFilename: string(identities)},
+	}
+
+	s.log.Info(fmt.Sprintf("Creating Operator Identities Secret %s", s.infinispan.GetSecretName()))
+	_, err = controllerutil.CreateOrUpdate(ctx, s.client, secret, func() error {
+		return controllerutil.SetControllerReference(s.infinispan, secret, s.scheme)
```

Comment:
This will be updated each Infinispan CR reconcile loop and every Infinispan CR will try to owned this resource
It would be better to have personal admin Secret for each cluster